apps/test/src/components/PositionsSection.tsx

```diff
@@ -95,6 +87,51 @@ const LBRow = (props: LBRowProps) => {
   );
 };
 
+interface RepayButtonProps {
+  leverageBuy: LeverageBuy;
+}
+
+const RepayButton = (props: RepayButtonProps) => {
+  const { leverageBuy } = props;
+  // deployment object, grabbed from the DeploymentProvider that wraps the app
+  const deployment = useDeployment();
+  // current connected signer
+  const { data: signer } = useSigner();
+  // wagmi query client
+  const { queryClient } = useClient();
+  // is repay transaction currently executing?
+  const [loading, setLoading] = useState(false);
+
+  // called when the repay button is clicked, it initiates a transaction to repay the loan
+  const repay = async () => {
+    if (!signer) throw new Error("repay called without a signer");
+    if (!deployment) throw new Error("repay was called without a deployment");
+    setLoading(true);
+    try {
+      // send tx
+      const tx = await repayETH({
+        escrowID: leverageBuy.escrowID,
+        repayment: leverageBuy.repayment,
+        signer,
+        lbWrapperAddress: deployment.lbWrapperAddress,
+      });
+      // wait for block confirmations
+      await tx.wait(2);
+      // wait for subgraph sync
+      if (tx.blockNumber) {
+        await waitForSubgraphSync({ blocNumber: tx.blockNumber, subgraphURI: deployment.subgraphURI });
+      }
+      // invalidate leverage buys query
+      queryClient.invalidateQueries(useLeverageBuys.queryKeys.all());
+    } catch {
+      // show a toast or something
+    }
+    setLoading(false);
+  };
```

Comment:
Did you mean a finally block?

Reply:
not really, why?

Reply:
I see, this is a void function. In this case there's no benefit 👍 
If it were to change you'd have to look out for it though, a return statement in the try block would cause the setLoading to not run.